src/api/firebase.js

```diff
@@ -87,10 +87,56 @@ export async function updateItem(listId, itemId, checked) {
 	const listItemRef = doc(db, listId, itemId);
 	const listItemSnap = await getDoc(listItemRef);
 
+	// Retrieving values stored in firebase
 	let totalPurchases = listItemSnap.data().totalPurchases;
+	let dateLastPurchased = listItemSnap.data().dateLastPurchased;
+	let dateCreated = listItemSnap.data().dateCreated;
+	let dateNextPurchased = listItemSnap.data().dateNextPurchased;
 
+	// Function estimating previous purchase interval
+	// Because our lists have updated dateLastPurchased without updated dateNextPurchased,
+	// the time intervals won't be accurate, so we recommend using only:
+	const temporaryPreviousEstimate = getDaysBetweenDates(
+		dateNextPurchased,
+		dateCreated,
+	);
+
+	// *** ONCE DATABASE IS CLEANED UP ***
+	// But longer term, if we hadn't been updating dateLastPurchased, this should be the long term logic:
+	// const previousEstimate = dateLastPurchased
+	// ? getDaysBetweeenDates(dateNextPurchased, dateLastPurchased)
+	// : getDaysBetweenDates(dateNextPurchased, dateCreated)
+
+	// This declares the number of days since the last transaction based on dateLastPurchased being a null or existing value
+	const numDaysSinceLastTransaction = dateLastPurchased
+		? // If dateLastPurchased returns TRUE, we get number of days between *dateLastPurchased* and current date
+		  getDaysBetweenDates(new Date(), dateLastPurchased.toDate())
+		: // If dateLastPurchased returns FALSE, we get number of days between *dateCreated* and current date
+		  getDaysBetweenDates(new Date(), dateCreated.toDate());
+
+	// Declaring variables for number of days until the NEXT purchase and converting to milliseconds
+	const numDaysUntilNextPurchase = calculateEstimate(
+		temporaryPreviousEstimate,
+		numDaysSinceLastTransaction,
+		totalPurchases,
+	);
+	const millisecondsUntilNextPurchase =
+		numDaysUntilNextPurchase * (24 * 60 * 60 * 1000);
+	// This is the nextPurchaseDate adding the the current date with the estimated next purchase date
+	const nextPurchaseDate = new Date(
+		new Date().getTime() + millisecondsUntilNextPurchase,
+	);
```

Comment:
This was very helpful when we were refactoring our code today! This is part of what we were asking during office hours too - which is how to make it our calculations easier to digest. We've updated the code so that `getNextPurchaseDate()` is a function that includes the `const nextPurchaseDate `similarly formatted here, but we ended up importing `ONE_DAY_IN_MILLISECONDS` from `dates.js`